### ISSUE TITLE
Linking error during MA5 installation: MA5-ERROR: impossible to link the project

### ISSUE THREAD

**Issue by apiccone15** [MadAnalysis/madanalysis5#99]
### Question

OS: Linux 3.10.0-1160.66.1.el7.x86_64
Python: Python 2.7.16
gcc: 8.3.0
MA5: 1.9.60 (also >=v1.6)

Hi, I've been trying to install a newer version of MA5 to interface with MG5 v2.6.5. After having a working installation of MA5 v1.6, I found that it was not directly compatible with my version of MG5 and opted to update MA5 manually (updating through MG5 has not been working properly). However, after removing v1.6 and installing 1.7 to see if this would fix compatibility with MG5, I began to run into this error upon first run of ./bin/ma5: 

MA5-ERROR: impossible to link the project. For more details, see the log file:
MA5-ERROR: /misc/raid03/users/alecp1/MG5_aMC_v2_6_5/HEPTools/madanalysis5/madanalysis5/tools/SampleAnalyzer/Test//linking_root.log
MA5-ERROR: The test program building aborted.

Components 1-8/11 while building the sample libraries all succeeded, except for this, the 9th (test program: interface to Root). This problem persists though all major releases of MA5 I've tried, even when retrying an installation of v1.6 which had initially worked. It also persists when vetoing root in installation_options.dat. The contents of the linking_root.log are: 

g++ Root/Test.o  -L/misc/raid03/users/alecp1/MG5_aMC_v2_6_5/HEPTools/madanalysis5/madanalysis5/tools/SampleAnalyzer/Li
b -L/misc/raid03/users/alecp1/MG5_aMC_v2_6_5/HEPTools/madanalysis5/madanalysis5/tools/SampleAnalyzer/ExternalSymLink/L
ib -lroot_for_ma5 -L/cvmfs/[sft.cern.ch/lcg/releases/ROOT/v6.20.06-3f7fd/x86_64-centos7-gcc8-opt/lib](http://sft.cern.ch/lcg/releases/ROOT/v6.20.06-3f7fd/x86_64-centos7-gcc8-opt/lib) -lCore -lImt -lRIO
 -lNet -lHist -lGraf -lGraf3d -lGpad -lROOTVecOps -lTree -lTreePlayer -lRint -lPostscript -lMatrix -lPhysics -lMathCor
e -lThread -lMultiProc -lROOTDataFrame -pthread -lm -ldl -rdynamic -lEG -lcommons_for_ma5 -L/cvmfs/sft.cern.ch/lcg/rel
eases/ROOT/v6.20.06-3f7fd/x86_64-centos7-gcc8-opt/lib -lCore -lImt -lRIO -lNet -lHist -lGraf -lGraf3d -lGpad -lROOTVec
Ops -lTree -lTreePlayer -lRint -lPostscript -lMatrix -lPhysics -lMathCore -lThread -lMultiProc -lROOTDataFrame -pthrea
d -lm -ldl -rdynamic -lEG -o ../Bin/TestRoot
/cvmfs/[sft.cern.ch/lcg/releases/ROOT/v6.20.06-3f7fd/x86_64-centos7-gcc8-opt/lib](http://sft.cern.ch/lcg/releases/ROOT/v6.20.06-3f7fd/x86_64-centos7-gcc8-opt/lib)/libImt.so: undefined reference to `tbb
::interface7::internal::task_arena_base::internal_terminate()'
/cvmfs/[sft.cern.ch/lcg/releases/ROOT/v6.20.06-3f7fd/x86_64-centos7-gcc8-opt/lib/libImt.so](http://sft.cern.ch/lcg/releases/ROOT/v6.20.06-3f7fd/x86_64-centos7-gcc8-opt/lib/libImt.so): undefined reference to `tbb
::interface7::internal::isolate_within_arena(tbb::interface7::internal::delegate_base&, long)'
/cvmfs/sft.cern.ch/lcg/releases/ROOT/v6.20.06-3f7fd/x86_64-centos7-gcc8-opt/lib/libImt.so: undefined reference to `tbb
::interface7::internal::task_arena_base::internal_initialize()'
/cvmfs/[sft.cern.ch/lcg/releases/ROOT/v6.20.06-3f7fd/x86_64-centos7-gcc8-opt/lib/libImt.so](http://sft.cern.ch/lcg/releases/ROOT/v6.20.06-3f7fd/x86_64-centos7-gcc8-opt/lib/libImt.so): undefined reference to `tbb
::interface7::internal::task_arena_base::internal_execute(tbb::interface7::internal::delegate_base&) const'
collect2: error: ld returned 1 exit status
make: *** [link] Error 1

If it helps any, during the first (working) installation of v1.6, I had to install a number of packages using MA5, but even after removing the entire directory, subsequent installations maintained that those packages still existed, which I found odd.

I'm guessing this has something to do with some sort of persistent environment variable or something to that tune, but I'm not sure what it could be, and my experience in unix-systems is very limited.  Any thoughts?

**Comment by jackaraz** [comment]
Hi @apiccone15, first of all, v1.6 is ancient. We have implemented many updates over v1.6 following the updates in ROOT's environment. I see that you are using ROOT v6.20. Honestly, I would be surprised if that ROOT version will even work with anything below Ma5-v1.9. 

Could you give me a bit more details here? Let's try it with Ma5-v1.9.60 and go step by step. I'm assuming that you would like to use Delphes with Ma5, and that's why you want to have ROOT? So I suggest a fresh installation of v1.9.6 which can be found [here](https://github.com/MadAnalysis/madanalysis5/releases). And please install all the packages you need through MadAnalysis this will ensure the installation of the compatible versions. Not every version of third-party software is compatible with MadAnalysis. Please first compile Ma5 without any third-party software and then, step by step, install all the packages you need and let me know if they break at any stage.

**Comment by apiccone15** [comment]
Hi Jack, 

I tried the fresh installation but ended up getting the same result as above (the problem ended up getting resolved, but I've attached the terminal outputs here in case they're of any interest). I took some time to try a few different things and found while running MA5 in debug that it was recognizing packages from the Views distribution that I have my bashrc setup and launch. I found that if I commented out that section of my bashrc and ran MA5 without Views interfering, I could download all the packages I needed, re-enable Views initialization in the bashrc and then re-source it without running into any trouble. It can still be finicky when trying to install new packages with Views initialized, but now I have a workaround, and as of now everything seems like it's working fine. Thanks!

[MA5freshinstallerror.txt](https://github.com/MadAnalysis/madanalysis5/files/8987262/MA5freshinstallerror.txt)
------------------------------------------------------------
[DebugOutput.txt](https://github.com/MadAnalysis/madanalysis5/files/8987261/DebugOutput.txt)
